modules/ddb_cover_upload/includes/ddb_cover_upload.submitted.inc

```diff
@@ -0,0 +1,107 @@
+<?php
+
+/**
+ * @file
+ * Cover upload page when form is submitted.
+ */
+
+use CoverServiceUpload\ApiException;
+use CoverServiceUpload\Api\CoverApi;
+use CoverServiceUpload\Model\MaterialWrite;
+use CoverServiceUpload\Api\MaterialApi;
+use GuzzleHttp\Client;
+
+/**
+ * Check cover service to determine if file has been uploaded.
+ *
+ * Ajax callback function.
+ */
+function _cover_upload_check_submission() {
+  $data = _ddb_cover_upload_session('ddb_cover_upload_submitted');
+
+  // Validation to prevent page re-load submit and upload.
+  $exists = FALSE;
+  if (isset($data['image_uri'])) {
+    $exists = file_exists(drupal_realpath($data['image_uri']));
+  }
+
+  // Check if the minimum required data exists.
+  if (!$exists || !isset($data['image_uri']) || !isset($data['pid'])) {
+    drupal_json_output(array(
+      'message' => t('Either the image have been uploaded ot something is wrong with the session'),
+      'status' => 'error',
+    ));
+    drupal_exit();
+  }
+
+  try {
+    _ddb_cover_upload_client_upload($data['image_uri'], $data['pid'], 'pid');
```

Comment:
Have you considered writing to `watchdog` that a new cover have been submitted?

Reply:
No, as there is a lot of logging at the service. So added watchdog statement.